test/async.js

```diff
@@ -0,0 +1,39 @@
+"use strict"
+
+/* global Promise */
+
+var async = require("../async")
+
+describe("clean-assert/async", function () {
```

Comment:
Would you mind adding some tests asserting rejection for things like `async.equal(Promise.resolve(1), 2)`?

Also, could you assert that existing rejections are passed through untouched like in `async.equal(Promise.reject(new Error("foo")), 1)`?